internal/resolver/dns/dns_resolver_test.go

```diff
@@ -714,17 +714,17 @@ func testDNSResolver(t *testing.T) {
 		},
 		{
 			"srv.ipv4.multi.fake",
-			nil,
+			[]resolver.Address{},
 			generateSC("srv.ipv4.multi.fake"),
 		},
 		{
 			"srv.ipv6.single.fake",
-			nil,
+			[]resolver.Address{},
 			generateSC("srv.ipv6.single.fake"),
 		},
 		{
 			"srv.ipv6.multi.fake",
-			nil,
+			[]resolver.Address{},
```

Comment:
Why is this change required? We should probably leave it as is.

Reply:
The fake host resolver that's used in the tests returns empty slices for `*.fake` addresses and the test ends up comparing `reflect.DeepEqual([]string{}, nil)` that fails.

`net.LookupHost` guarantees that slice is not empty when error is nil but the test expects `state` to be updated with an empty list of addresses.

Reply:
I think we should keep this as `nil`, and change the `reflect.DeepEqual` to `cmp.Equal` to which you can pass a `cmpopts.EquateEmpty` which is meant exactly for this usecase.